sdk/aws/ec2.py

```diff
@@ -87,47 +98,158 @@ def list_instances(self, params_dict=None):
+                return {"count": 0, "instances": [], "error": "No subnets found."}
+            subnet_id = random.choice(subnet_ids)
+
+            # Fetch security group
+            security_group_id = self._get_security_group_id(sec_group_name="Allow SSH")
+            if not security_group_id:
+                logger.error("No Security Groups found.")
+                return {
+                    "count": 0,
+                    "instances": [],
+                    "error": "No Security Group found with name Allow SSH.",
+                }
+
+            # Define instance parameters
             instance_params = {
                 "ImageId": image_id,
                 "InstanceType": instance_type,
                 "KeyName": key_name,
                 "SecurityGroupIds": [security_group_id],
+                "SubnetId": subnet_id,
+                "TagSpecifications": [
+                    {
+                        "ResourceType": "instance",
+                        "Tags": [{"Key": "Name", "Value": f"{username}"}],
+                    }
+                ],
                 "MinCount": 1,
                 "MaxCount": 1,
             }
-            if subnet_id:
-                instance_params["SubnetId"] = subnet_id
 
-            instances = ec2.create_instances(**instance_params)
+            # Now create the EC2 instance using the defined parameters
+            instances = ec2_resource.create_instances(**instance_params)
 
             if not instances:
                 logger.warning(f"Unable to create EC2 instance: {instance_params}")
+                return {"count": 0, "instances": []}
 
-                return {
-                    "count": 0,
-                    "instances": [],
-                }
+            instance = instances[0]
+            instance.wait_until_running()
+            instance.reload()
 
-            server_name = instances[0].id
-            logger.info(f"Server {server_name} created successfully")
-            server_info = {
-                "name": server_name,
+            logger.info(
+                f"Instance {instance.id} created successfully with name '{username}'"
+            )
+
+            instance_info = {
+                "name": username,
+                "instance_id": instance.id,
                 "key_name": key_name,
                 "instance_type": instance_type,
+                "public_ip": instance.public_ip_address,
             }
 
             return {
                 "count": 1,
-                "instances": [server_info],
+                "instances": [instance_info],
             }
+
         except Exception as e:
-            logger.error(f"An error occurred creating the EC2 instance {e}")
-            raise e
+            logger.error(f"An error occurred creating the EC2 instance: {e}")
+            logger.debug(traceback.format_exc())
+            return {
+                "count": 0,
+                "instances": [],
+                "error": str(e),
```

Comment:
let's be careful to not return this error to the bot

Reply:
Yes I removed the say part that returns the error to bot so it will not display on the UI but in our logs